### DISCUSSION TITLE
Termux as a safe

### DISCUSSION THREAD

**Discussion by beroal** [termux/termux-app#2597] [General]
I discovered a use case for Termux which I wish to share. Some programs refuse to work if they aren't granted the access to the shared storage (`/storage/emulated/0`). (Fenix (Firefox) is an example.) In this case, you can store your personal documents in the Termux document provider. It exposes the app-specific storage of the Termux application which doesn't belong to the shared storage.

**Comment by SDRausty** [comment]
try storage permissions off

Android grants auto storage dirs

**Comment by agnostic-apollo** [comment]
Yeah, I use it too for stuff like KeePassDX. Scoped storage improves things in that department but only for apps targeting android 11 and on 10 not requesting legacy storage. But still better to keep some files safe from apps having `MANAGE_EXTERNAL_STORAGE` permission.
